### ISSUE TITLE
[yugabyted] Error while starting yugabyte cluster with './yugabyted start' on macOS

### ISSUE THREAD

**Issue by gargsans-yb** [yugabyte/yugabyte-db#13373]
### Description

When trying to start a 1-node cluster on macOS using `./yugabyted start`, following error shows up.
```
gargsans@Sanskars-MacBook-Pro bin % ./yugabyted start
Traceback (most recent call last):
  File "./yugabyted", line 3723, in <module>
    ControlScript().run()
  File "./yugabyted", line 2250, in run
    self.validate_and_set_configs(args)
  File "./yugabyted", line 1976, in validate_and_set_configs
    args.advertise_address = self.find_ip_address_of_node()
  File "./yugabyted", line 1845, in find_ip_address_of_node
    host_ip = socket.gethostbyname(host_name)
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```
The python method socket.gethostbyname() is not able to get the host ip.

**Comment by nchandrappa** [comment]
Another issue reported for itest failure for a default behaviour change. We will need update the tests before landing this feature.

we have reverted the code change in master. The issue is test case is trying to connect on 127.0.0.1 by default however a feature was landed to change the behavior of yugabyted to bind on the host VM private-ip address instead on 127.0.0.1. YCQLSH should use host IP instead of loopback address. 

 

The test case has to be updated however this feature was broken on macOS, so we reverted the feature. There are workarounds for this issue. Do we have to backport the reverted code changes for 2.15.1.0 branch? Please let us know

**Comment by Yosuamuliawan19** [comment]
Hi, are there any updates on this ? Im currently unable to start installing yugabtyeDB on my local machine due to this. Machine: Intel MBP 15 inch , Mac OS Big Sur

**Comment by nchandrappa** [comment]
We have reverted the code changes from the master, and a new fix is in progress. The issue will be fixed in YugabyteDB 
2.15.2 release. Thanks

**Comment by nchandrappa** [comment]
Hi @Yosuamuliawan19 There is a workaround by directly specifying the advertise_address. You can run the following command to start a single node cluster - 
```
./bin/yugabyted start --adverstise_address=127.0.0.1
```

**Comment by nchandrappa** [comment]
Code changes have landed and they will be available in `2.15.3.0` release.

**Comment by dpkirchner** [comment]
I can confirm this error still happens in 2.25.1.0-b381, so I guess maybe there was a regression.
